src/middleware/TestingHandler.ts

```diff
@@ -0,0 +1,265 @@
+
+		if (statusCode === 301 || statusCode === 302 || statusCode === 303 || statusCode === 307 || statusCode === 308) {
+			// adding location header for only these cases as done for the redirect handler
+			responseHeader.Location = this.redirectURL;
+		}
+
+		if (statusCode === 429) {
+			// throttling case has to have a timeout scenario
+			responseHeader["retry-after"] = 300;
+		}
+		return responseHeader;
+	}
+
+	/**
+	 * @private
+	 * Generates responseBody
+	 * @param {number} statusCode - the status code to be returned for the request
+	 * @param {string} statusMessage - the status message to be returned for the request
+	 * @param {string} requestID - request id
+	 * @param {Date} errDate - date of the request
+	 * @returns response body
+	 */
+	private createResponseBody(statusCode: number, statusMessage: string, requestID: string, requestDate: Date) {
+		// response body gets created as empty for passed cases
+		// response body contains error field for failure scenarios
+		let responseBody;
+		if (statusCode >= 400) {
+			const codeMessage: string = httpStatusCode[statusCode];
+			const errMessage: string = statusMessage;
+
+			responseBody = {
+				error: {
+					code: codeMessage,
+					message: errMessage,
+					innerError: {
+						"request-id": requestID,
+						date: requestDate,
+					},
+				},
+			};
+		} else {
+			responseBody = {};
+		}
+
+		return responseBody;
+	}
+
+	/**
+	 * @private
+	 * creates a response object out of responseHeader and responseBody
+	 * @param {TestingHandlerOptions} testingHandlerOptions - The TestingHandlerOptions object
+	 * @param {string} requestURL - the URL for the request
+	 * @returns Response object
+	 */
+	private createResponse(testingHandlerOptions: TestingHandlerOptions, requestURL: string, requestMethod: string): Response {
+		try {
+			// creates a response Object out of responseHeader and responseBody
+			let responseBody;
+			let responseHeader;
+			let requestID: string;
+			let requestDate: Date;
+
+			this.setStatusCode(testingHandlerOptions, requestURL, requestMethod);
+			requestID = generateUUID();
+			requestDate = new Date();
+			responseHeader = this.createResponseHeaders(testingHandlerOptions.statusCode, testingHandlerOptions.statusMessage, requestID, requestDate);
+			responseBody = this.createResponseBody(testingHandlerOptions.statusCode, testingHandlerOptions.statusMessage, requestID, requestDate);
+			const init = { url: requestURL, status: testingHandlerOptions.statusCode, statusText: testingHandlerOptions.statusMessage, headers: responseHeader };
+			const response = new Response(responseBody, init);
+
+			return response;
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	/**
+	 * @private
+	 * Fetches a random status code for the RANDOM mode from the predefined array
+	 * @param {string} requestMethod - the API method for the request
+	 * @returns the random status code
+	 */
+	private getRandomStatusCode(requestMethod: string): number {
```

Comment:
change request method type from string to RequestMethod which is defined in the file RequestMethod.ts

Reply:
this applies to all the places where the requestMethod is used